woocommerce/woocommerce
# Marketplace: Add no results page

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:
When we have no results as a result of a search term or a category selection, we show this message and the most popular products section. 

### How to test the changes in this Pull Request:

1.Build WooCommerce Admin on your local environment: `cd plugins/woocommerce-admin` then `nvm use && pnpm start`.
2. Visit WooCommerce > Marketplace > Extensions (eg. http://localhost:8888/wp-admin/admin.php?page=wc-admin&path=%2Fextensions&tab=extensions).
3. Search for some term (for example shipping or tax) and see the products listed on the page.
4. Search for some term that won't result in any results and see the new no results page.
5. Remove your search term and select the **Featured** category. **Featured** category, somehow doesn't have any results. See the error message now saying "We didn't find any results for "_featured". This is not a great message and I'm thinking about a possible solution to this.
    - I'm annoyed about the `_featured` bit in the message. The fact that it starts with an underscore makes it look bad to my eyes.

![image (8)](https://github.com/woocommerce/woocommerce/assets/10389957/53ed89aa-43b1-4417-8af0-09ade02359b8)
